tests/area/test_coefficient_area.py

```diff
@@ -0,0 +1,161 @@
+        house1 = grid_area.children[0]
+        house2 = grid_area.children[1]
+        time_slot = now()
+        scm = SCMManager(grid_area.uuid, time_slot)
+        grid_area.calculate_home_after_meter_data(time_slot, scm)
+        scm.calculate_community_after_meter_data()
+
+        assert scm._home_data[house1.uuid].community_production_kWh == 0.1
+        assert isclose(scm._home_data[house1.uuid].allocated_community_energy_kWh, 0.06)
+        assert isclose(scm._home_data[house1.uuid].energy_bought_from_community_kWh, 0.06)
+        assert isclose(scm._home_data[house1.uuid].energy_sold_to_grid_kWh, 0.0)
+
+        assert scm._home_data[house2.uuid].community_production_kWh == 0.1
+        assert isclose(scm._home_data[house2.uuid].allocated_community_energy_kWh, 0.04)
+        assert isclose(scm._home_data[house2.uuid].energy_bought_from_community_kWh, 0.00)
+        assert isclose(scm._home_data[house2.uuid].energy_sold_to_grid_kWh, 0.04)
+
+    @classmethod
+    def test_trigger_energy_trades(cls):
+        grid_area = cls._create_2_house_grid()
+        house1 = grid_area.children[0]
+        house2 = grid_area.children[1]
+        time_slot = now()
+        scm = SCMManager(grid_area.uuid, time_slot)
+        grid_area.calculate_home_after_meter_data(time_slot, scm)
+        scm.calculate_community_after_meter_data()
+        grid_area.trigger_energy_trades(scm)
+
+        # energy need * normal market maker fees for the case of positive energy need
+        assert isclose(scm._bills[house1.uuid].base_energy_bill, 0.06)
+        assert isclose(scm._bills[house1.uuid].gsy_energy_bill, 0.06)
+        assert isclose(scm._bills[house1.uuid].savings_euro, 0.0)
+        assert isclose(scm._bills[house1.uuid].savings_percent, 0.0)
+        # energy surplus * feed in tariff for the case of positive energy surplus
+        assert isclose(scm._bills[house2.uuid].base_energy_bill, -0.005)
+        assert isclose(scm._bills[house2.uuid].gsy_energy_bill, -0.0164)
+        assert isclose(scm._bills[house2.uuid].savings_euro, 0.0114)
+        assert isclose(scm._bills[house2.uuid].savings_percent, -228.0)
+        assert len(scm._home_data[house1.uuid].trades) == 2
+        trades = scm._home_data[house1.uuid].trades
+        assert isclose(trades[0].trade_rate, 0.3)
+        assert isclose(trades[0].traded_energy, 0.06)
+        assert trades[0].seller == "Community"
+        assert trades[0].buyer == "House 1"
+        assert isclose(trades[1].trade_rate, 0.3)
+        assert isclose(trades[1].traded_energy, 0.14)
+        assert trades[1].seller == "Grid"
+        assert trades[1].buyer == "House 1"
+        assert len(scm._home_data[house2.uuid].trades) == 2
+        trades = scm._home_data[house2.uuid].trades
+        assert isclose(trades[0].trade_rate, 0.24)
+        assert isclose(trades[0].traded_energy, 0.06)
+        assert trades[0].seller == "Community"
+        assert trades[0].buyer == "House 2"
```

Comment:
Why is the "House 2" buying energy? AFAIU, it does not need to sell energy, but needs to sell the energy, that it received from the Community (0.04 kWh) to the grid again.

Reply:
Ah yes, of course! Thanks!